docs/operate/deploy-with-terraform.mdx

```diff
@@ -0,0 +1,207 @@
+- Creates an ECS service to manage the deployment and scaling of the Validator tasks.
+
+## Main Configuration
+
+The root level configuration sets up a VPC, subnets, internet gateway, NAT gateway, route tables, and security groups for network infrastructure. It also provides example usage of the `validator` module.
+
+```terraform
+module "your_validator_name" {
+  source = "./modules/validator"
+
+  validator_name    = "your-validator-name"
+  origin_chain_name = "originChainName"
+
+  aws_region               = var.aws_region
+  validator_cluster_id     = aws_ecs_cluster.validator_cluster.id
+  validator_subnet_id      = aws_subnet.validator_subnet.id
+  validator_sg_id          = aws_security_group.validator_sg.id
+  validator_nat_gateway_id = aws_nat_gateway.validator_nat_gateway.id
+
+  # Disabling the Validator task allows you to set up all the required infrastructure
+  # without running the actual Validator yet. This is useful when setting up a Validator for
+  # the first time, so that you can find out the Validator's address and fund it before it
+  # performs the announcement transaction.
+  # validator_task_disabled = true
+}
+```
+
+### Outputs
+
+The root-level `outputs.tf` passes forward all outputs from the Validators configured in `main.tf`. **You will have to update this as you add, modify or remove Validators.**
+
+### Example architecture
+
+The diagram below shows how the Validator ECS cluster fits into the top-level network infrastructure.
+
+```mermaid
+graph TD
+    subgraph VPC_Network ["VPC Network"]
+        VPC[AWS VPC]
+        Public_Subnet[AWS Public Subnet]
+        Private_Subnet[AWS Private Subnet]
+        VPC -- Connected to --> IGW[AWS Internet Gateway]
+        VPC -- Contains --> SG[AWS Security Group]
+        Public_Subnet -- Internet Access via --> IGW
+        Private_Subnet -- Internet Access via NAT --> NAT_Gateway[AWS NAT Gateway]
+        NAT_Gateway -- Utilizes --> EIP[AWS Elastic IP]
+        NAT_Gateway -- Located in --> Public_Subnet
+
+        subgraph Routing ["Routing"]
+            Public_Subnet -- Associates --> Public_Route_Table[AWS Public Route Table]
+            Public_Route_Table -- Routes Internet Traffic --> IGW
+            Private_Subnet -- Associates --> Private_Route_Table[AWS Private Route Table]
+            Private_Route_Table -- Routes Traffic To --> NAT_Gateway
+        end
+    end
+
+    subgraph ECS_Infrastructure ["ECS Infrastructure"]
+        VPC -- Hosts --> ECS_Cluster[AWS ECS Cluster]
+        SG -- Secures --> ECS_Cluster
+    end
+
+    %% Additional Details
+    SG -- Secures --> Public_Subnet
+    SG -- Secures --> Private_Subnet
+
+    %% Internet Node
+    Internet[Internet]
+    IGW -- Connects to --> Internet
+    EIP -- Connects to --> Internet
+```
+
+## Known issues
+
+### Non-core configuration
+
+It is non-trivial to set custom configuration values for non-core chains.
```

Comment:
do you think "non-core chain" is obvious to someone new to Hyperlane? wondering if "a chain you've deployed Hyperlane to yourself" or something is more palatable. Don't feel strongly tho, happy to leave as is too

Reply:
I think if you've made it to this point in the guide you should know what it is, but we've not used "non-core" as a term before, will change to your suggestion to make it more obvious